src/scripts/main.js

```diff
@@ -1,3 +1,67 @@
 'use strict';
 
-// write code here
+const btnAddRow = document.querySelector('.append-row');
+const btnDelRow = document.querySelector('.remove-row');
+const btnAddCol = document.querySelector('.append-column');
+const btnDelCol = document.querySelector('.remove-column');
+const tbody = document.querySelector('tbody');
+
+document.querySelector('.container').addEventListener('click', (e) => {
+  const target = e.target;
+  const minNumbersBlocks = 2;
+  const maxNumbersBlocks = 10;
+
+  switch (target) {
+    case btnAddRow:
+      addRow(tbody);
+      break;
+
+    case btnDelRow:
+      delRow(tbody.children);
+      break;
+
+    case btnAddCol:
+      addColumn(tbody.children);
+      break;
+
+    case btnDelCol:
+      delColumn(tbody);
+      break;
+  }
+
+  tbody.children.length >= maxNumbersBlocks
+    ? btnAddRow.disabled = true
+    : btnAddRow.disabled = false;
+
+  tbody.firstChild.children.length <= minNumbersBlocks
+    ? btnDelRow.disabled = true
+    : btnDelRow.disabled = false;
+
+  tbody.firstChild.children.length >= maxNumbersBlocks
+    ? btnAddCol.disabled = true
+    : btnAddCol.disabled = false;
+
```

Comment:
Still not resolved for these cases 
<img width="850" alt="Screenshot 2022-12-06 at 19 59 33" src="https://user-images.githubusercontent.com/104555497/205961195-0d9666b1-2093-46a4-85b9-9db835c40406.png">